x/distribution/types/genesis.go

```diff
@@ -67,3 +71,30 @@ func DefaultGenesisWithValidators(valAddrs []sdk.ValAddress) GenesisState {
 		DelegationDistInfos: ddis,
 	}
 }
+
+// ValidateGenesis validates the genesis state of distribution genesis input
+func ValidateGenesis(data GenesisState) error {
+	if data.CommunityTax.IsNegative() {
+		return fmt.Errorf("mint parameter CommunityTax should be positive, is %s",
+			data.CommunityTax.String())
+	}
+	if data.CommunityTax.GT(sdk.OneDec()) {
+		return fmt.Errorf("mint parameter CommunityTax should be less than one, is %s",
+			data.CommunityTax.String())
+	}
+	if data.BaseProposerReward.IsNegative() {
```

Comment:
Have a felling `BaseProposerReward` is also supposed to be `0 < BaseProposerReward < 1`. The next case here is a replica. Might want it to look like:

```go
if data.BaseProposerReward.GT(sdk.OneDec()) || data.BaseProposerReward.IsNegative()  {
	return fmt.Errorf("mint parameter BaseProposerReward should be positive and less than one, is %s",data.BaseProposerReward.String())
}
```

Reply:
nah, the `>1` case is covered two below, slightly different logic it's:
```
BaseProposerReward + BonusProposerReward > 1
```